jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -117,10 +112,15 @@ def __init__(self, *args, **kwargs):
                                                      on_rename=self._on_overlay_rename,
                                                      on_remove=self._on_overlay_remove)
 
+        if preset_regions._has_pysiaf:
+            preset_options = list(preset_regions._instruments.keys())
+        else:
+            preset_options = []
+        preset_options.append('From File...')

```

Comment:
this won't work as-is because it will immediately raise the file dialog as soon as the plugin is rendered.  Something like this is definitely an option down the road, but since the goal is to move the functionality of pysiaf to be internal soon, I don't think it makes sense to do that yet.

If you insist in order to test, I can work on something along these lines, but we will also need to display messages in the app explaining why the JWST footprints are not available (similar to the disabled message, but while still providing the from file functionality).

Reply:
Either that or we make a new plugin. I feel like it's important for region import to be useful beyond JWST footprint. If we just build on this now and defer everything to later, it's just another refactoring waiting to happen.

Reply:
We have a planned work coming up very soon to attempt to move the pysiaf logic locally.  If for some reason that isn't feasible, then I agree that we should only disable those features of the plugin and not the whole thing, but it will take some additional logic than just what you have here (although, granted, probably not much more).  Ideally one of those two things would happen before this finds its way in a release 🤞 

But we also have to keep in mind that this plugin's primary use is for instrument footprints.  So I agree, if we want more generic regions support, we should either rename it to be more general or have a separate more-general plugin.

Reply:
I don't think it is crazy to have a dialog pop up not to be automatic, i.e., move it to a new upload button to the side that someone needs to press manually after the dropdown is selected. Like the layout in launcher.